code/game/objects/machinery/squad_supply/supply_console.dm

```diff
@@ -0,0 +1,179 @@
+	. = ..()
+	if(.)
+		return
+
+	switch(action)
+		if("select_squad")
+			var/datum/squad/selected_squad = SSjob.squads[params["select_squad"]]
+			if(!istype(selected_squad))
+				return
+			current_squad = selected_squad
+			refresh_squad_pad()
+
+		if("set_x")
+			var/new_x = text2num(params["set_x"])
+			if(!new_x)
+				return
+			x_offset = new_x
+
+		if("set_y")
+			var/new_y = text2num(params["set_y"])
+			if(!new_y)
+				return
+			y_offset = new_y
+
+		if("refresh_pad")
+			refresh_squad_pad()
+
+		if("send_beacon")
+			if(next_fire > world.time)
+				return
+
+			if(!current_squad?.sbeacon)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>There was an issue with that squads beacon. Check it's still active.</span>")
+				return
+
+			if(!length(supplies))
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>There wasn't any supplies found on the squads supply pad. Double check the pad.</span>")
+				return
+
+			var/area/A = get_area(current_squad.sbeacon)
+			if(A && A.ceiling >= CEILING_DEEP_UNDERGROUND)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>The [current_squad.sbeacon.name]'s signal is too weak. It is probably deep underground.</span>")
+				return
+
+			var/turf/T = get_turf(current_squad.sbeacon)
+			if(!istype(T))
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>The [current_squad.sbeacon.name] was not detected on the ground.</span>")
+				return
+			if(isspaceturf(T) || T.density)
+				to_chat(usr, "[icon2html(src, usr)] <span class='warning'>The [current_squad.sbeacon.name]'s landing zone appears to be obstructed or out of bounds.</span>")
+				return
+
+			next_fire = world.time + launch_cooldown
+			current_squad.send_supplydrop(supplies, x_offset, y_offset)
+			to_chat(world, "Beacon sent to [current_squad.sbeacon]")
+
+
+/obj/machinery/computer/supplydrop_console/proc/refresh_squad_pad()
+	supplies = list()
+	for(var/obj/C in current_squad.drop_pad.loc) //This thing should ALWAYS exist.
+		if(is_type_in_typecache(C, GLOB.supply_drops) && !C.anchored) //Can only send vendors and crates
+			supplies.Add(C)
+		if(supplies.len > MAX_SUPPLY_DROPS)
+			break
+
+/datum/squad/proc/send_supplydrop(list/supplies, x_offset = 0, y_offset = 0)
+	if(!sbeacon)
+		stack_trace("Trying to send a supply drop without a squad beacon")
+		return
+
+	if(!length(supplies) || length(supplies) > MAX_SUPPLY_DROPS)
+		stack_trace("Trying to send a supply drop with an invalid amount of items [length(supplies)]")
+		return
+
+	var/turf/T = get_turf(sbeacon)
+	if(!istype(T) || isspaceturf(T) || T.density)
+		stack_trace("Trying to send a supply drop to a beacon on an invalid turf")
+		return
+
+	x_offset = clamp(round(x_offset), -5, 5)
+	y_offset = clamp(round(y_offset), -5, 5)
+
+	drop_pad.visible_message("<span class='boldnotice'>The supply drop is now loading into the launch tube! Stand by!</span>")
```

Comment:
Shouldnt this be a tochat for the user?

Reply:
This gives it runechat and looks cool. Personally think its better.